exercises/fulfill_a_promise/problem.md

```diff
@@ -1,50 +1,41 @@
 # What is a promise?
-
-A promise is an object that defines a method called `then`.
-The promise object represents a value (or values) that may
-be available some time in the future.
-
-When a promise is resolved, any *"success functions"* registered
-with the `then` method will be called with the newly available
-data as arguments.
-
-If a promise is rejected then any *"failure functions"* registered
-with the `then` method will be called with the `Error` as argument.
-
-For now, do not be concerned with exactly **how** this works or what
-the syntax is.  We are about to dive into that in detail.
+A promise is an object that lets a consumer attach success and failure
+handlers to an asyncrhonous operation. If the operation succeeds,
+the function you register using the promises `then` function
+will be called with the result of the operation. If the operation
+fails, the function you register with the promises
+`catch` function is called.
```

Comment:
Okay, if I recall correctly though this koan later asks you to use catch.
So I was attempting to introduce it early. Certainly not hard to change
that.

On Fri, Apr 24, 2015 at 1:00 PM, Timothy Gu <notifications@github.com>
wrote:

> In exercises/fulfill_a_promise/problem.md
> <https://github.com/stevekane/promise-it-wont-hurt/pull/54#discussion_r29065711>
> :
>
> > -
> > -When a promise is resolved, any *"success functions"* registered
> > -with the `then` method will be called with the newly available
> > -data as arguments.
> > -
> > -If a promise is rejected then any *"failure functions"* registered
> > -with the `then` method will be called with the `Error` as argument.
> > -
> > -For now, do not be concerned with exactly **how** this works or what
> > -the syntax is.  We are about to dive into that in detail.
> > +A promise is an object that lets a consumer attach success and failure
> > +handlers to an asyncrhonous operation. If the operation succeeds,
> > +the function you register using the promises `then` function
> > +will be called with the result of the operation. If the operation
> > +fails, the function you register with the promises
> > +`catch` function is called.
>
> .catch is non-standard.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevekane/promise-it-wont-hurt/pull/54/files#r29065711>
> .
>